Admin/Admin.php

```diff
@@ -2540,8 +2546,12 @@ public function id($entity)
     /**
      * {@inheritdoc}
      */
-    public function setValidator(ValidatorInterface $validator)
+    public function setValidator($validator)
```

Comment:
But that would mean being able to modify the argument before to wrap it, and I don't think we can do that.